File: python/04_013.py
from abc import ABC, abstractmethod

class Document(ABC):
    @abstractmethod
    def open(self):
        pass

class PDFDocument(Document):
    def open(self):
        # PDF specific logic
        pass

class DocumentFactory:
    def create_document(self, doc_type):
        if doc_type == "pdf":
            return PDFDocument()
        # Add other document types
        return None

File: data structures and algorithms/java/04_013.java
public interface Document {
    void open();
}

public class PDFDocument implements Document {
    public void open() { /* PDF specific logic */ }
}

public class DocumentFactory {
    public Document createDocument(String type) {
        if (type.equals("pdf")) {
            return new PDFDocument();
        }
        // Add other document types
        return null;
    }
}